homeassistant/helpers/config_validation.py

```diff
@@ -1745,18 +1745,35 @@ def expand_condition_shorthand(value: Any | None) -> Any:
     "numeric_state": NUMERIC_STATE_CONDITION_SCHEMA,
     "or": OR_CONDITION_SCHEMA,
     "state": STATE_CONDITION_SCHEMA,
-    "sun": SUN_CONDITION_SCHEMA,
     "template": TEMPLATE_CONDITION_SCHEMA,
     "time": TIME_CONDITION_SCHEMA,
     "trigger": TRIGGER_CONDITION_SCHEMA,
     "zone": ZONE_CONDITION_SCHEMA,
 }
 
+
+# This is first round of validation, we don't want to mutate the config here already,
+# just ensure basics as condition type and alias are there.
+def _base_condition_validator(value: Any) -> Any:
+    vol.Schema(
+        {
+            **CONDITION_BASE_SCHEMA,
+            CONF_CONDITION: vol.NotIn(BUILT_IN_CONDITIONS),
```

Comment:
I don't understand why we have this validation, ie not a built in condition.

Reply:
Without that, an invalid built-in condition would pass this schema.
However, I think it should be changed to `vol.All(str, vol.NotIn(BUILT_IN_CONDITIONS))` or something like that

Reply:
If/when all conditions are moved to condition platforms, the `BUILT_IN_CONDITIONS` would go away, unless we want to keep the validation in the `config_validation` helper for some reason.